### ISSUE TITLE
Confusing error for `if let ... && ...`

### ISSUE THREAD

**Issue by camelid** [rust-lang/rust#78203]
```rust
fn main() {
    if let x = 3 && true {}
}
```

([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c0d118d6227b2b3a849013d47eb4115a))

I think only the first error should be reported:

```
   Compiling playground v0.0.1 (/playground)
error[E0658]: `let` expressions in this position are experimental
 --> src/main.rs:2:8
  |
2 |     if let x = 3 && true {}
  |        ^^^^^^^^^
  |
  = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information

error: expected expression, found statement (`let`)
 --> src/main.rs:2:8
  |
2 |     if let x = 3 && true {}
  |        ^^^^^^^^^
  |
  = note: variable declaration using `let` is a statement

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0658`.
error: could not compile `playground`.

To learn more, run the command again with --verbose.
```

**Comment by c410-f3r** [comment]
This issue can be closed

**Comment by camelid** [comment]
Yep, see also https://github.com/rust-lang/rust/issues/78205#issuecomment-1017914994.
